components/event-service/internal/events/subscribed/events.go

```diff
@@ -36,13 +43,51 @@ type eventsClient struct {
 	triggerLister kneventinglister.TriggerLister
 }
 
+// waitForInformersSyncOrDie blocks until all informer caches are synced, or panics after a timeout.
+func waitForInformersSyncOrDie(f externalversions.SharedInformerFactory) {
+	ctx, cancel := context.WithTimeout(context.Background(), informerSyncTimeout)
+	defer cancel()
+
+	err := hasSynced(ctx, f.WaitForCacheSync)
+	if err != nil {
+		log.Fatalf("Error waiting for caches sync: %s", err)
+	}
+}
+
+type waitForCacheSyncFunc func(stopCh <-chan struct{}) map[reflect.Type]bool
+
+// hasSynced blocks until the given informer sync waiting function completes. It returns an error if the passed context
+// gets canceled.
+func hasSynced(ctx context.Context, fn waitForCacheSyncFunc) error {
+	// synced gets closed as soon as fn returns
+	synced := make(chan struct{})
+
+	// closing stopWait forces fn to return, which happens whenever ctx
+	// gets canceled
+	stopWait := make(chan struct{})
+	defer close(stopWait)
+
+	go func() {
+		fn(stopWait)
+		close(synced)
+	}()
+
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case <-synced:
+	}
+
+	return nil
+}
+
 //NewEventsClient function creates client for retrieving all active events
 func NewEventsClient(client versioned.Interface) EventsClient {
 	informerFactory := externalversions.NewSharedInformerFactory(client, 0)
 	lister := informerFactory.Eventing().V1alpha1().Triggers().Lister()
 	ctx := signals.NewContext()
 	informerFactory.Start(ctx.Done())
-	informerFactory.WaitForCacheSync(ctx.Done())
+	waitForInformersSyncOrDie(informerFactory)
```

Comment:
can we add a log that we're waiting for cache to sync, ideally it'd be good to see in the log if this service is blocked on waiting for cache to sync